### PR TITLE
v1.13 Backports 2024-06-25

### PR DESCRIPTION
**Pull request by YutaroHayakawa**
* [ ] #26779 (@ferozsalam)
 * [x] #32951 (@giorio94) :warning: resolved conflicts
 * [x] #33216 (@aanm)
 * [ ] #33218 (@dependabot[bot]) :warning: resolved conflicts
 * [x] #33353 (@aanm)
 * [ ] #33304 (@learnitall)
 * [x] #33368 (@aanm)

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 26779 32951 33216 33218 33353 33304 33368
```

### CODE REVIEW

File: pkg/client/client.go [hunk 1]
```diff
@@ -382,7 +382,7 @@ func FormatStatusResponse(w io.Writer, sr *models.StatusResponse, sd StatusDetai
 	}
 
 	if sr.ClusterMesh != nil {
-		fmt.Fprintf(w, "ClusterMesh:\t%d/%d clusters ready, %d global-services\n",
+		fmt.Fprintf(w, "ClusterMesh:\t%d/%d remote clusters ready, %d global-services\n",
 			numReadyClusters(sr.ClusterMesh), len(sr.ClusterMesh.Clusters), sr.ClusterMesh.NumGlobalServices)
 
 		for _, cluster := range sr.ClusterMesh.Clusters {
```
Comment (giorio94) [inline pkg/client/client.go:388]: The word `failures` should be changed to `reconnections` at line 390 (sorry, I cannot comment there).

File: pkg/client/client.go [hunk 2]
```diff
@@ -708,6 +708,6 @@ func FormatStatusResponse(w io.Writer, sr *models.StatusResponse, sd StatusDetai
 			fields = append(fields, fmt.Sprintf("[%s]", strings.Join(ifaces, ", ")))
 		}
 
-		fmt.Fprintf(w, "Encryption:\t%s\n", strings.Join(fields, "\t"))
+		fmt.Fprintf(w, "Encryption:\t%s\t%s\n", sr.Encryption.Mode, strings.Join(fields, ", "))
```
Comment (giorio94) [inline pkg/client/client.go]: This change should be reverted, as not present in the original commit.

File: Documentation/operations/troubleshooting_clustermesh.rst [hunk 3]
```diff
@@ -63,8 +63,8 @@ you may perform the following steps to troubleshoot ClusterMesh issues.
 
  #. Validate that ClusterMesh is healthy running ``cilium status --all-clusters`` inside each Cilium agent::
 
-        ClusterMesh:   1/1 clusters ready, 10 global-services
-           k8s-c2: ready, 3 nodes, 8 identities, 10 services, 0 failures (last: never)
+        ClusterMesh:   1/1 remote clusters ready, 10 global-services
+           k8s-c2: ready, 3 nodes, 25 endpoints, 8 identities, 10 services, 0 reconnections (last: never)
```
Comment (giorio94) [inline Documentation/operations/troubleshooting_clustermesh.rst]: The `25 endpoints, ` change should be reverted (while preserving the `reconnections` one).

**Review by giorio94 — CHANGES_REQUESTED** [review]
Thanks, and sorry for the conflicts that you had to resolve. Just a couple of minor comments inline.

File: pkg/client/client.go (hunk 1 above)
Comment (YutaroHayakawa) [reply pkg/client/client.go:388]: Thanks! Should be fixed now with other ones.

**Review by giorio94 — APPROVED** [review]
My commits look good, thanks!

**Review by ferozsalam — APPROVED** [review]

**Comment by YutaroHayakawa** [comment]
/test-backport-1.13

**Review by aanm — APPROVED** [review]

**Comment by YutaroHayakawa** [comment]
/test-backport-1.13

**Review by learnitall — APPROVED** [review]
Thanks!

**Comment by YutaroHayakawa** [comment]
Conformance AKS: bing.com timeout
Conformance Cluster Mesh: google.com timeout
Conformance E2E: one.one.one.one timeout

**Comment by YutaroHayakawa** [comment]
/test-1.17-4.19

**Review by viktor-kurchenko — APPROVED** [review]

**Review by nathanjsweet — APPROVED** [review]

**Review by youngnick — APPROVED** [review]

**Comment by YutaroHayakawa** [comment]
/test-1.17-4.19

**Comment by YutaroHayakawa** [comment]
Hmm, looks like the retry is not triggered?

**Comment by YutaroHayakawa** [comment]
/test-1.17-4.19

**Pull request merged**